src/components/lyrics.tsx

```diff
@@ -1,74 +1,88 @@
 import { useEffect, useState } from "react";
 
-export default function Lyrics({ lyricsLink }: { lyricsLink: string }) {
-  const [translatedLyrics, setTranslatedLyrics] = useState(null);
-  const [lyrics, setLyrics] = useState(null);
+const fetchLyrics = async (lyricsLink: string) => {
+  try {
+    const response = await fetch(
+      `/api/lyrics/${encodeURIComponent(lyricsLink)}`
+    );
+    if (!response.ok) {
+      throw new Error(`Response status: ${response.status}`);
+    }
+    const data = await response.json();
+    return data.lyrics;
+  } catch (error) {
+    console.error("Error fetching lyrics:", error);
+  }
+};
 
-    const fetchLyrics = async () => {
-        try {
-          const response = await fetch(`/api/lyrics/${encodeURIComponent(lyricsLink)}`);
-          if (!response.ok) {
-            throw new Error(`Response status: ${response.status}`);
-          }
-    
-          const data = await response.json();
-          if(data.lyrics){
-          setLyrics(data.lyrics);
-          await translateLyrics(data.lyrics);
-        } 
-        } catch (error) {
-          console.error("Error fetching lyrics:", error);
-        }
-      };
-    
+const translateLyrics = async (lyrics: string) => {
+  try {
+    const response = await fetch("/api/translate", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        text: lyrics,
+        targetLang: "pl",
+      }),
+    });
 
-  const translateLyrics = async (lyrics: string) => {
-    try {
-      const response = await fetch("/api/translate", {
-        method: "POST",
-        headers: {
-          "Content-Type": "application/json",
-        },
-        body: JSON.stringify({
-          text: lyrics,
-          targetLang: "pl",
-        }),
-      });
-  
-      const data = await response.json();
-  
-      if (data.translatedText) {
-        setTranslatedLyrics(data.translatedText);
-      } 
-    } catch (error) {
-      console.error("Error translating lyrics:", error);
-    }
-  };
+    const data = await response.json();
+    return data.translatedText;
+  } catch (error) {
+    console.error("Error translating lyrics:", error);
+  }
+};
+
+export default function Lyrics({ lyricsLink }: { lyricsLink: string }) {
+  const [translatedLyrics, setTranslatedLyrics] = useState<string | null>(null);
+  const [lyrics, setLyrics] = useState<string | null>(null);
+  const [isLoadingLyrics, setIsLoadingLyrics] = useState(true);
+  const [isLoadingTranslatedLyrics, setIsLoadingTranslatedLyrics] = useState(true);
+  const [requestMade, setRequestMade] = useState(false);
 
   useEffect(() => {
+
+    const makeRequest = async () => {
+      const lyrics = await fetchLyrics(lyricsLink);

```

Comment:
on start of every fetch you have to set loading to true, because when you change for example lyricsLink, useEffect runs, but loading still false, so you wont see loader, only old text until new not fetched